python_modules/dagster/dagster/_core/definitions/assets.py

```diff
@@ -1344,17 +1344,21 @@ def _subset_graph_backed_asset(
                 "Method _subset_graph_backed_asset cannot subset an asset that is not a graph"
             )
 
-        # All asset keys in selected_asset_keys are outputted from the same top-level graph backed asset
-        dep_node_handles_by_asset_or_check_key = self.dep_op_handles_by_asset_or_check_key
+        dep_op_handles_by_asset_or_check_key = cast(
+            # because self.node_def is a graph, these NodeHandles that reference ops inside it will
+            # not be None
+            Mapping["AssetKeyOrCheckKey", AbstractSet[NodeHandle]],
+            self.dep_op_handles_by_asset_or_check_key,
+        )
         op_selection: List[str] = []
         for asset_key in selected_asset_keys:
-            dep_node_handles = dep_node_handles_by_asset_or_check_key[asset_key]
```

Comment:
Took the liberty of changing "node" to "op" in these names to make it clear that we're only targeting leaf nodes.